converter/src/main/java/org/kaaproject/avro/ui/converter/SchemaFormAvroConverter.java

```diff
@@ -455,8 +455,16 @@ private Record createTypeFromSchema(Schema schema, Set<String> namedFqns) {
                         GenericData.Array<Object> fieldsArrayData = new GenericData.Array<>(fields.size(), fieldsField.schema());
                         
                         Schema fieldSchema = fieldsField.schema().getElementType();
+                        java.util.Set<String> fieldNames = new java.util.HashSet<>();
                         
                         for (Field field : fields) {
+                            String fieldName = field.name().toLowerCase();
+                            if (!fieldNames.contains(fieldName)) {
+                                fieldNames.add(fieldName);
+                            } else {
+                                throw new RuntimeException("Field names are not unique!");
+                            }
+
                             fieldsArrayData.add(createFormFieldFromSchemaField(fieldSchema, field, namedFqns));
```

Comment:
Change RuntimeException to IllegalArgumentException. Change exception message to indicate which field is duplicate.